sdk/ai/Azure.AI.Inference/src/Customized/AIInferenceExtensions.cs

```diff
@@ -17,47 +19,81 @@ public static class AIInferenceExtensions
+        public static ChatCompletionsClient GetChatCompletionsClient(this ConnectionProvider provider, string? connectionName=default, AzureAIInferenceClientOptions? options = default)
         {
-            ChatCompletionsClient chatClient = provider.Subclients.GetClient(() => CreateChatCompletionsClient(provider), null);
+            ChatCompletionsClient chatClient = provider.Subclients.GetClient(() => CreateChatCompletionsClient(provider, connectionName, options), null);
             return chatClient;
         }
 
-        private static ChatCompletionsClient CreateChatCompletionsClient(this ConnectionProvider provider)
+        private static ChatCompletionsClient CreateChatCompletionsClient(this ConnectionProvider provider, string? connectionName, AzureAIInferenceClientOptions? options)
         {
-            ClientConnection connection = provider.GetConnection(typeof(ChatCompletionsClient).FullName!);
+            ClientConnection connection = GetConnnectionByNameOrDefault(provider, connectionName, typeof(ChatCompletionsClient).FullName!);
+
             if (!connection.TryGetLocatorAsUri(out Uri? uri) || uri is null)
             {
                 throw new InvalidOperationException("Invalid URI.");
             }
             return connection.Authentication == ClientAuthenticationMethod.Credential
-            ? new ChatCompletionsClient(uri, connection.Credential as TokenCredential)
-            : new ChatCompletionsClient(uri, new AzureKeyCredential(connection.ApiKeyCredential!));
+            ? new ChatCompletionsClient(uri, connection.Credential as TokenCredential, ["https://cognitiveservices.azure.com/.default"], options)
+            : new ChatCompletionsClient(uri, new AzureKeyCredential(connection.ApiKeyCredential!), options);
+        }
+
+        /// <summary>
+        /// Get the default connection if connectionName is null and get connection with given name otherwise.
+        /// If no Connections with the given name were found, Throw InvalidOperationException.
+        /// </summary>
+        /// <param name="provider">The connection provider to be extended by this method.</param>
+        /// <param name="connectionName">The name of connection to search for</param>
+        /// <param name="connectionType">The type of a connection; ignored if connectionName is set.</param>
+        /// <returns></returns>
+        private static ClientConnection GetConnnectionByNameOrDefault(this ConnectionProvider provider, string? connectionName, string connectionType)
+        {
+            ClientConnection connection = default;
+            if (string.IsNullOrEmpty(connectionName))
+            {
+                // Get Default connection for given service type.
+                connection = provider.GetConnection(connectionType);
+            }
+            else
+            {
+                IEnumerable<ClientConnection> conns = provider.GetAllConnections().Where(
```

Comment:
this is not how we use the connectionIds in other clients that support multiple clients of the same type. We simply append additional disambiguating info to the client name. See https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/cloudmachine/Azure.Projects/src/extensions/BlobExtensions.cs#L27.